sdk/v2/core/projects.go

```diff
@@ -85,6 +86,18 @@ func (p ProjectList) MarshalJSON() ([]byte, error) {
 	)
 }
 
+// ValidateProjectID checks if a given id is valid.
+func ValidateProjectID(id string) error {
+	regex := regexp.MustCompile(`^[a-z][a-z\d-]*[a-z\d]$`)
+	idMatch := regex.MatchString(id)
+	if !idMatch || len(id) < 3 || len(id) > 63 {
+		return fmt.Errorf("invalid value %q for --id flag"+
+			" (3-63 characters, first char must be"+
+			" a letter, lowercase letters only)", id)
+	}
+	return nil
+}
+
```

Comment:
Let's move this down to the bottom of the file, please, so it's set apart from API-related stuff.